### ISSUE TITLE
[$250] Room -System message with members invited, is not updated on #Admins until opening room

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53915]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.74-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5325392&group_by=cases:section_id&group_order=asc&group_id=229067
**Email or phone of affected tester (no customers):** ibellicotest+89@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Open the staging.new.expensify.com website.
2. Open any room chat or create one if necessary.
3. Send a mention including some users that are not part of the room.
4. When a whisper message with options appears, select "Invite Them" on it.
5. Return to LHN.
6. Verify that in #Admins room preview, a message with this last action is seen.
7. Open the #Admins room and verify that a system message is displayed reflecting the last action made on the room chat.

## Expected Result:
A system message should be displayed on #Admins room, everytime that a user is invited or removed from a room.

## Actual Result:
When inviting users to a room via whisper, the system message reflecting this action is not displayed on #Admins room until opening it.

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/6f2ecf24-1fb7-47f4-8df3-2422e1c858f7

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021869832623281675425</li>
        <li>Upwork Job ID: 1869832623281675425</li>
        <li>Last Price Increase: 2025-02-06</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @suneox</details>

**Comment by slafortune** [comment]
@daledah the two invited users do show in the "test" room,but not indicated correctly in the Admin room.

**Comment by suneox** [comment]
This issue still reproduces on the latest main branch `v9.0.78-2` and we're still looking for proposals. 
It appears to occur only after opening the #admin room.

https://github.com/user-attachments/assets/1867f2ae-72e9-4451-9398-90a83ac78ba8

**Comment by linhvovan29546** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When inviting users to a room via whisper, the system message reflecting this action is not displayed on #Admins room until opening it.
### What is the root cause of that problem?
When inviting users to a room via [whisper](https://github.com/Expensify/App/blob/918488aa5586be249f392b26692572aaba010f31/src/libs/actions/Report.ts#L4175), we do not build `optimisticData` for the admin room, which causes the system message not to display until the admin room is opened

https://github.com/Expensify/App/blob/918488aa5586be249f392b26692572aaba010f31/src/libs/actions/Report.ts#L4203-L4221

### What changes do you think we should make in order to solve the problem?
We need to build `optimisticData` and `failureData` (to revert to the last message of the admin room as we do with the report room)  for `report` and `report_actions` of the admin room, we also need to back-end support the parameters for `report_actions` of the admin room


Example mock:
```js
    const policy = getPolicy(report?.policyID);
    const isAdmin = policy?.role === CONST.POLICY.ROLE.ADMIN;
    if (isAdmin) {
        //Build optimistic data
        const chatReportIDAdmins = policy.chatReportIDAdmins
        const adminReportActionOptimisticData = ReportUtils.buildOptimisticResolveActionableMentionWhisperAdminReportAction(report,policy)
        const adminReportOptimisticData= ReportUtils.buildOptimisticResolveActionableMentionWhisperAdminReport(report,policy)

        optimisticData.push(
            {
                onyxMethod: Onyx.METHOD.MERGE,
                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportId}`,
                value: {
                    [adminReportActionOptimisticData.reportActionID]: adminReportActionOptimisticData,
                },
            },
            {
                onyxMethod: Onyx.METHOD.MERGE,
                key: `${ONYXKEYS.COLLECTION.REPORT}${chatReportIDAdmins}`,
                value: adminReportOptimisticData,
            },
        )

        parameters = {
            ...parameters,
            adminReportActionID: adminReportAction.reportActionID
        }
    }
```
<details>
 <summary>POC</summary>

https://github.com/user-attachments/assets/346786b0-07b9-4178-8826-7b95f10f4337


</details>

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?
We can write a test in `OptionsListUtilsTest` to verify that the system message is displayed on LHN of the admin room.

### What alternative solutions did you explore? (Optional)
N/A

**Comment by slafortune** [comment]
@suneox do you agree this is a BE issue?

**Comment by suneox** [comment]
> @suneox do you agree this is a BE issue?

@slafortune  I don't think this is a BE issue, as the problem doesn't occur after we have a chat in the #admin room, and the BE response also includes an invited message.


https://github.com/user-attachments/assets/ffe37ce2-495d-4e93-8a78-40cbf9b9d552


@linhvovan29546 Thanks for your proposal. Could you please investigate the root cause of the BE response includes an invited message in the admin room but it is not displayed?

**Comment by linhvovan29546** [comment]
> @linhvovan29546 Thanks for your proposal. Could you please investigate the root cause of the BE response includes an invited message in the admin policy but it is not displayed?
The root cause is that the backend does return a message empty in the `reportActions` because we display the message on the LHN based on the `visibleReportActionItems`
<img width="887" alt="Screenshot 2024-12-31 at 8 48 57 AM" src="https://github.com/user-attachments/assets/8bf0cacc-d6ed-487d-9f4b-22f6cbef81f3" />

https://github.com/Expensify/App/blob/89215464ce7bff50c01f63aa34fc8785c6fd50f2/src/libs/SidebarUtils.ts#L59-L71

Then the LHN will display the previous message.
https://github.com/Expensify/App/blob/89215464ce7bff50c01f63aa34fc8785c6fd50f2/src/libs/SidebarUtils.ts#L441

It seems we do not need to build `optimisticData` for `admin` room in this case.

**Comment by suneox** [comment]
> The root cause is that the backend does return a message empty in the reportActions

@linhvovan29546  Thanks for your update, but the RCA is still incorrect and this issue cannot be reproduced on the latest staging `v9.0.80-2` without the message you mentioned.

https://github.com/user-attachments/assets/9cc96c0e-4907-4a69-9d4e-569df0b0cc1e

@slafortune This issue cannot be reproduced on the latest staging `v9.0.80-2` so we can add the retest label for this one

**Comment by linhvovan29546** [comment]
<img width="917" alt="Screenshot 2025-01-01 at 10 51 59 PM" src="https://github.com/user-attachments/assets/84497aff-d30b-4489-90f7-a8a043089c04" />
I still reproduce that on main branch

**Comment by suneox** [comment]
This issue still cannot be reproduced on the latest staging version `v9.0.82-1`, so we can add the “retest” label to this issue.

https://github.com/user-attachments/assets/a8137725-f195-47a9-8a5b-0a899678be5b

**Comment by slafortune** [comment]
Thanks @suneox  - done 👍

**Comment by slafortune** [comment]
Updated to weekly for testing

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by slafortune** [comment]
waiting on another test

**Comment by slafortune** [comment]
Waiting on another test - hopefully we can close this then

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)
